metadata-ingestion/tests/integration/powerbi/test_powerbi.py

```diff
@@ -1191,3 +1191,117 @@ def test_server_to_platform_map(
     # to process all available upstream lineage even if mapping for platform instance is
     # not provided in server_to_platform_instance map
     dataset_type_mapping_set_to_all_platform(pipeline)
+
+
+@freeze_time(FROZEN_TIME)
+@mock.patch("msal.ConfidentialClientApplication", side_effect=mock_msal_cca)
+@pytest.mark.integration
+def test_pages_by_report(mock_msal, pytestconfig, tmp_path, mock_time, requests_mock):
```

Comment:
Can we remove these test-case and add a singe unit-test case where we will call pwerbi_api.py get_reports function and assert the return response, so lets say there four reports and for one reports we are raising 400 (mock data) then the function should return only three reports